packages/cli/test/integration/user.repository.test.ts

```diff
@@ -38,4 +40,25 @@ describe('UserRepository', () => {
 			});
 		});
 	});
+
+	describe('createUserWithProject()', () => {
+		test('should create personal project for a user', async () => {
+			const user = await userRepository.createUserWithProject({
+				email: randomEmail(),
+				role: 'global:member',
+			});
+
+			const projectRelation = await Container.get(ProjectRelationRepository).findOneOrFail({
+				where: {
+					userId: user.id,
+					project: {
+						type: 'personal',
+					},
+				},
+				relations: ['project'],
+			});
+
+			expect(projectRelation.project).not.toBeUndefined();
```

Comment:
I'd rather check that it's the project ID we expect.

Reply:
How would you do that? The project gets a random ID assigned:
![image](https://github.com/n8n-io/n8n/assets/927609/bd2ee0f2-ffce-409a-ac1f-f1e55abd8039)

Reply:
Even if the ID is random, if we create a project during a test, I'd expect we should be able to return it after creation and refer to its ID in the assertion.

Reply:
I changed the return type of `UserRepository.createUserWithProject` to also create the project that has been created. That way the test can assert that when using the relation to look up the project it finds the same project.

[`722915b` (#8550)](https://github.com/n8n-io/n8n/pull/8550/commits/722915b86a812a51475eadc6aba4d1e4e75d8b8a)

Is that what you had in mind?

Reply:
Exactly, thank you :)

Can we remove the `.only` in the owner tests?

Reply:
ooops, yes

Reply:
I removed the whole test. I didn't want to include it. The owner project is created by the migration not by the setup routes. So it's a bit weird to test that the project exists when testing the route 🤷🏾